src/anemoi/graphs/inspector.py

```diff
@@ -0,0 +1,200 @@
+                    number(nodes.x[:, 0].max().item() / 2 / math.pi * 360),
+                    number(nodes.x[:, 1].min().item() / 2 / math.pi * 360),
+                    number(nodes.x[:, 1].max().item() / 2 / math.pi * 360),
+                ]
+            )
+        return node_summary
+
+    def get_edge_summary(self) -> list[list]:
+        """Summary of the edges in the graph.
+
+        Returns
+        -------
+        list[list]
+            Returns a list for each subgraph with the following information:
+            - Source node name.
+            - Destination node name.
+            - Number of edges.
+            - Number of isolated source nodes.
+            - Number of isolated target nodes.
+            - Total dimension of the attributes.
+            - List of attribute names.
+        """
+        edge_summary = []
+        for (src_name, _, dst_name), edges in self.graph.edge_items():
+            attributes = edges.edge_attrs()
+            attributes.remove("edge_index")
+
+            edge_summary.append(
+                [
+                    src_name,
+                    dst_name,
+                    number(edges.num_edges),
+                    number(self.graph[src_name].num_nodes - len(torch.unique(edges.edge_index[0]))),
+                    number(self.graph[dst_name].num_nodes - len(torch.unique(edges.edge_index[1]))),
+                    sum(edges[attr].shape[1] for attr in attributes),
+                    ", ".join(attributes),
+                ]
+            )
+        return edge_summary
+
+    def describe(self) -> None:
+        """Describe the graph."""
+        print()
+        print(f"📦 Path       : {self.path}")
+        print(f"💽 Size       : {bytes(self.total_size)} ({number(self.total_size)})")
+        print()
+        print(
+            table(
+                self.get_node_summary(),
+                header=[
+                    "Nodes name",
+                    "Num. nodes",
+                    "Attributes",
+                    "Attribute dim",
+                    "Min. latitude",
+                    "Max. latitude",
+                    "Min. longitude",
+                    "Max. longitude",
+                ],
+                align=["<", ">", ">", ">", ">", ">", ">", ">"],
+                margin=3,
+            )
+        )
+        print()
+        print()
+        print(
+            table(
+                self.get_edge_summary(),
+                header=[
+                    "Source",
+                    "Target",
+                    "Num. edges",
+                    "Isolated Source",
+                    "Isolated Target",
+                    "Attribute dim",
+                    "Attributes",
+                ],
+                align=["<", "<", ">", ">", ">", ">", ">"],
+                margin=3,
+            )
+        )
+        print("🔋 Graph ready.")
+        print()
+
+
+class GraphInspectorTool:
```

Comment:
I would try to name `GraphInspectorTool` and `GraphDescription` following the same way so either:
- GraphInspectorTool and GraphDescriptorTool
- GraphInspection and GraphDescription

(as possible suggestions)

Reply:
How about GraphInspector and GraphDescriptor?

Reply:
I like it!